src/Pass/SugarImgurTransformPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Lullabot\AMP\Pass;
+
+use QueryPath\DOMQuery;
+
+use Lullabot\AMP\Utility\ActionTakenLine;
+use Lullabot\AMP\Utility\ActionTakenType;
+
+/**
+ * Class SugarImgurPass
+ * @package Lullabot\AMP\Pass
+ */
+class SugarImgurTransformPass extends BasePass
+{
+    function pass()
+    {
+        $all_imgur = $this->q->top()->find('blockquote.imgur-embed-pub');
+        /** @var DOMQuery $el */
+        foreach ($all_imgur as $el) {
+            /** @var \DOMElement $dom_el */
+            $dom_el = $el->get(0);
+            $lineno = $this->getLineNo($dom_el);
+            $context_string = $this->getContextString($dom_el);
+
+            /** @var \DOMElement $new_dom_el */
+            $imgur_id = $el->attr('data-id');
+            $img_src = 'https://i.imgur.com/' . $imgur_id . '.jpg';
+            list($width, $height) = getimagesize($img_src);
+
+            if (!$width || !$height) {
+                $this->addActionTaken(new ActionTakenLine('blockquote.imgur', ActionTakenType::IMGUR_COULD_NOT_BE_CONVERTED, $lineno, $context_string));
+                continue;
+            }
+
+            $height += 150;
+
```

Comment:
It would be a lot easier to read if you built this string with HEREDOC:
```
$amp_string =<<<"HTML"
<amp-iframe 
                height="$height"
                width="$width"
...
HTML;

Then you can avoid all the back-slashes